### PR TITLE
added unit tests for join

### PR DESCRIPTION
**Pull request by hunterprice04**

### CODE REVIEW

File: tests/test_join.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    """
+
+    @classmethod
+    def setup_class(cls):
+        """
+        This method is run once before any of the class' test methods are run.
+        It sets up the environment for the tests, and checks that the token is provided and the API is up.
+        """
+
+        cls.base_url = 'http://localhost:5000'
```
Comment (andreicozma1) [inline tests/test_join.py:19]: is there any way to re-use some of this code that's duplicated in the other file?
Perhaps creating a file for these repeated functions that returns a tuple of the exact dependencies that need to be set on the cls method?

**Review by andreicozma1 — CHANGES_REQUESTED** [review]

File: tests/test_join.py (hunk 1 above)
Comment (hunterprice04) [reply tests/test_join.py:19]: scope creep, seperate issue made here https://cs401.atlassian.net/browse/BE-108

Comment (andreicozma1) [reply tests/test_join.py:19]: not sure what u mean by scope creep but avoiding the introduction of duplicated code is just following good programming practices for a solid implementation

Comment (hunterprice04) [reply tests/test_join.py:19]: Refactoring these setup calls is unrelated to testing joining groups. there is a separate issue that @tdbucks is currently working on that is a refactor of the tests and will contain this fix

Comment (andreicozma1) [reply tests/test_join.py:19]: Sounds good. If this is within the scope of a pre-existing issue, do we still need BE-108? At this point in time I say we should avoid creating more issues unless absolutely necessary

**Review by andreicozma1 — APPROVED** [review]

**Pull request merged**